src/source.ts

```diff
@@ -108,13 +106,37 @@ export class Source extends React.Component<Props> {
 
   public componentDidUpdate(prevProps: Props) {
     const { geoJsonSource, tileJsonSource, map } = prevProps;
+    const source = map.getSource(this.id);
 
     // Update tilesJsonSource
     if (tileJsonSource && this.props.tileJsonSource) {
+      let urlUpdated = false;
+      let tilesUpdated = false;
+
+      if (source && source.type === 'vector') {
+        const hasNewSourceUrl =
+          tileJsonSource.url !== this.props.tileJsonSource.url;
+
+        if (hasNewSourceUrl && this.props.tileJsonSource.url) {
```

Comment:
this doesn't permit unsetting the URL – not sure if there would be a valid use case for it, but is it intentional here?

Reply:
The Mapbox types seem to indicate that the url cannot be "unset", i.e the functions don't take undefined as an argument - for that I guess one would have to change the entire source. I did now however make this check more explicit about undefined.